_studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_ddi.cpp

```diff
@@ -429,10 +429,12 @@ mfxStatus FillCUQPDataDDI(Task& task, MfxVideoParam &par, VideoCORE& core, mfxFr
         FrameLocker lock(&core, task.m_midCUQp);
         MFX_CHECK(lock.Y, MFX_ERR_LOCK_MEMORY);
 
-        for (mfxU32 i = 0; i < CUQPFrameInfo.Height; i++)
+        for (mfxU32 i = 0; i < CUQPFrameInfo.Height; i++) {
             for (mfxU32 j = 0; j < CUQPFrameInfo.Width; j++)
-                    lock.Y[i * lock.Pitch + j] = mbqp->QP[i*drBlkH/inBlkSize * pitch_MBQP + j*drBlkW/inBlkSize];
-
+                lock.Y[i * lock.Pitch + j] = mbqp->QP[i*drBlkH / inBlkSize * pitch_MBQP + j * drBlkW / inBlkSize];
+            for (mfxU32 j = CUQPFrameInfo.Width; j < lock.Pitch; j++) // hevc averages to LCU size
```

Comment:
"hevc averages to LCU size" comment doesn't seem to be right. AFAIU, this new 'for' block populates last QP in a row to position [width, pitch]. So there is no averaging here.

Reply:
The comment answers to "what for" not "what is it".
When frame is unaligned to LCU, qp for all LCU blocks are averaged in hw, so these values have to be set. Averaging is only horizontal here, btw.

Reply:
When I suggest to emphasize in comment that averaging is done in HW

Reply:
Added to commit comment.

Reply:
I mean code comment "// hevc averages to LCU size"

Reply:
Please check